### PR TITLE
feat(odd): Large Button

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-351

# Overview

Creates component, story and test for Large Button in ODD

# Test Plan

examine the 3 types of buttons + the disabled state on storybook, should match figma

# Changelog

- create ODD `LargeButton` component, test and storybook

# Review requests

- see test plan

# Risk assessment

low

### CODE REVIEW

**Review by ewagoner — APPROVED** [review]

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const LARGE_BUTTON_STYLE = css`
+    text-align: ${TYPOGRAPHY.textAlignLeft};
+    color: ${LARGE_BUTTON_PROPS_BY_TYPE[buttonType].defaultColor};
+    background-color: ${LARGE_BUTTON_PROPS_BY_TYPE[buttonType]
+      .defaultBackgroundColor};
+    cursor: default;
+    border-radius: ${BORDERS.size_four};
+    box-shadow: none;
+    padding: ${SPACING.spacing5} ${SPACING.spacing5} 2.4375rem
+      ${SPACING.spacing5};
```
Comment (koji) [inline app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: top | left & right | bottom

```suggestion
    padding: ${SPACING.spacing5} ${SPACING.spacing5} 2.4375rem;
```

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    <NewPrimaryBtn
+      {...buttonProps}
+      css={LARGE_BUTTON_STYLE}
+      aria-label={`LargeButton_${buttonType}`}
+      flexDirection={DIRECTION_ROW}
+    >
+      <StyledText
+        fontSize="2rem"
+        fontWeight={TYPOGRAPHY.fontWeightSemiBold}
+        paddingBottom="4.6875rem"
```
Comment (koji) [inline app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx:112]: seems that StyledText is missing `lineHeight`

```suggestion
        lineHeight="2.625rem"
        paddingBottom="4.6875rem"
```

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+} from '@opentrons/components'
+import { StyledText } from '../../text'
+import type { StyleProps } from '@opentrons/components'
+
+type LargeButtonTypes = 'primary' | 'secondary' | 'alert'
+interface LargeButtonProps extends StyleProps {
+  onClick: () => void
+  buttonType: LargeButtonTypes
+  buttonText: React.ReactNode
+  disabled?: boolean
```
Comment (koji) [inline app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx:22]: In the Context section, this large button component uses different icons for each button.
I think we need to pass IconName as a props.

```suggestion
  disabled?: boolean
  iconName: IconName
```

**Review by koji — COMMENTED** [review]
Mostly looks good to me but from the Figma page, we will need to pass an icon name.

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx (hunk 3 above)
Comment (jerader) [reply app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx:22]: oops good catch! I will add that, thank you!

**Review by koji — APPROVED** [review]
Tested passing an iconName to this component and it worked as expected.
⏺️

**Pull request merged**